src/forms/subscriptions.ts

```diff
@@ -1,18 +1,21 @@
 import {Channel} from 'mattermost-redux/types/channels';
-import {AppSelectOption, AppCall, AppForm, AppField} from 'mattermost-redux/types/apps';
+import {AppSelectOption, AppCallRequest, AppForm, AppField} from 'mattermost-redux/types/apps';
+import {AppContextWithBot} from 'types/apps';
 import {AppFieldTypes} from 'mattermost-redux/constants/apps';
 import Client4 from 'mattermost-redux/client/client4.js';
 
 import {newZDClient, newMMClient, ZDClient} from '../clients';
 import {getStaticURL, Routes} from '../utils';
-import {getBulletedList, makeSubscriptionOptions, makeChannelOptions, getChannelIDFromTriggerTitle, tryPromiseWithMessage} from '../utils/utils';
+import {getBulletedList, makeSubscriptionOptions, makeChannelOptions, getChannelIDFromTriggerTitle, tryPromiseWithMessage, ZDFieldOption} from '../utils/utils';
+import {ZDTrigger, ZDTriggerCondition} from '../utils/ZDTypes';
 import {SubscriptionFields, ZendeskIcon} from '../utils/constants';
 import {BaseFormFields} from '../utils/base_form_fields';
 
 // newSubscriptionsForm returns a form response to create subscriptions
-export async function newSubscriptionsForm(call: AppCall): Promise<AppForm> {
-    const zdClient = await newZDClient(call.context);
-    const mmClient = newMMClient(call.context).asAdmin();
+export async function newSubscriptionsForm(call: AppCallRequest): Promise<AppForm> {
+    const context: AppContextWithBot = call.context as AppContextWithBot;
```

Comment:
Either declare the type or use `as`. No need to use both

Reply:
done.